tests/session/session-pin-001.phpt

```diff
@@ -0,0 +1,56 @@
+--TEST--
+MongoDB\Driver\Session test: executeBulkWrite pins transaction to server
```

Comment:
I think we should refer to `Manager::executeBulkWrite()` here, since we're not testing the Server method. On that note, I think we'd benefit from testing that the Server methods also pin, too. That would also mean we could add error tests for Server methods to show that they still pass their own `server_id` to libmongoc instead of using a sessions pinned server. We can assert libmongoc's error message for those cases.

Rather than cram all of these into `session-pin-xxx`, what do you think about adding them under the actual method being tested? General convention for tests is to name things after the method under test, but we do deviate in some cases where it's ambiguous (e.g. APM tests, comprehensive cursor tests).

Reply:
I've moved the tests to the `server` and `manager` directories, testing each command, query, and bulkWrite methods in each.

I also followed your suggestion to add error tests for server methods to assert that the commands error out. I did not split these tests off in separate tests, as we'd be duplicating a lot of code to set up the pinned session. Let me know if that's ok with you or if you'd rather keep the two cases separate.

Testing the error case was a good call on your part, as this uncovered a case where libmongoc does not initialise the BSON reply object, causing an invalid call to `bson_destroy` in the PHP driver (see [trace in travis-ci](https://travis-ci.org/mongodb/mongo-php-driver/jobs/576703164#L1261)). I've reported this to the C team as [CDRIVER-3336](https://jira.mongodb.org/browse/CDRIVER-3336).